frontend/comment/comment.component.js

```diff
@@ -0,0 +1,190 @@
+                .then(function sucess() {
+                    commentCtrl.addLike();
+                    commentCtrl.saving = false;
+                }, function error() {
+                    $state.go("app.user.home");
+                    commentCtrl.saving = false;
+                });
+        };
+
+        commentCtrl.dislike = function () {
+            commentCtrl.saving = true;
+            CommentService.dislike(commentCtrl.post.key, commentCtrl.comment.id, commentCtrl.replyId)
+                .then(function sucess() {
+                    commentCtrl.removeLike();
+                    commentCtrl.saving = false;
+                }, function error() {
+                    $state.go("app.user.home");
+                    commentCtrl.saving = false;
+                });
+        };
+
+        commentCtrl.replyComment = function replyComment() {
+            if (commentCtrl.newReply) {
+                commentCtrl.saving = true;
+                var institutionKey = commentCtrl.user.current_institution.key;
+                CommentService.replyComment(
+                    commentCtrl.post.key, commentCtrl.newReply, institutionKey, commentCtrl.comment.id
+                ).then(function success(data) {
+                    commentCtrl.comment.replies[data.id] = data;
+                    commentCtrl.newReply = null;
+                    commentCtrl.saving = false;
+                }, function error() {
+                    commentCtrl.newReply = null;
+                    commentCtrl.saving = false;
+                });
+            }
+        };
+
+        commentCtrl.deleteReply = function deleteReply() {
+            CommentService.deleteReply(commentCtrl.post.key, commentCtrl.comment.id, commentCtrl.replyId)
+                .then(function success() {
+                    delete commentCtrl.comment.replies[commentCtrl.replyId];
+                    MessageService.showToast('Comentário excluído com sucesso');
+                });
+        };
+
+        commentCtrl.deleteComment = function deleteComment() {
+            CommentService.deleteComment(commentCtrl.post.key, commentCtrl.comment.id).then(
+                function success() {
+                    commentCtrl.post.data_comments = commentCtrl.post.data_comments
+                        .filter(comment => comment.id !== commentCtrl.comment.id);
+                    commentCtrl.post.number_of_comments--;
+                    MessageService.showToast('Comentário excluído com sucesso');
+                });
+        };
+
+        commentCtrl.confirmCommentDeletion = function confirmCommentDeletion(event) {
+            MessageService.showConfirmationDialog(event, 'Excluir Comentário',
+                'Este comentário será excluído e desaparecerá do referente post.'
+            ).then(function () {
+                commentCtrl.reply ? commentCtrl.deleteReply() : commentCtrl.deleteComment();
```

Comment:
what about the component has an input `onDelete` and you pass the correct function.

Reply:
I did not create a new input, instead of that, I added the property onDelete to the comment controller, so its reference changes to the deleteComment function or to the deleteReply function, depending on whether the component is loading a comment or a reply.